### PR TITLE
Use webpack-dev-middleware function waitUntilValid to resolve promise

### PR DESCRIPTION
**Pull request by christensson**
Solves issue #2 for me at least...

Room for further improvement: reject promise on errors...

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -57,13 +57,8 @@ function koaDevware(compiler, options) {
 
   function middleware(context, next) {
     return new _promise2.default(function (resolve, reject) {
-      // https://github.com/webpack/docs/wiki/plugins#donestats-stats
-      compiler.plugin('done', function (stats) {
-        resolve(stats);
-      });
-
-      compiler.plugin('failed', function (error) {
```
Comment (shellscape) [inline dist/index.js]: why remove the failed handler?

Comment (christensson) [inline dist/index.js]: Perhaps I shouldn't have removed that one, but still more error-handling is needed. For example the case where a non-existing resource is requested. Then the promise needs to be resolved/rejected.

Comment (shellscape) [inline dist/index.js]: a non-existing resource in which context? an endpoint/file on the server, or a resource within a file on the server?

Comment (christensson) [inline dist/index.js]: Let's say that I setup the webpack configuration to dynamically serve /js/bundle.js which is referenced from index.html which is statically served from another middleware. I found out that if I do a curl for localhost:<somePort>/js/bundle2.js which webpack isn't setup to dynamically serve nothing happens, no response is returned, no 404, no nothing....

Comment (shellscape) [inline dist/index.js]: that's really interesting. that needs to be logged as another issue.

**Comment by christensson** [comment]
done

**Comment by shellscape** [comment]
ah still got the changes to /dist in that commit. can you remove that file? I want to get the other issue fixed before update the dist file.

**Comment by christensson** [comment]
@shellscape Yes you do, but the union of the 2 commits in this pull-request doesn't... The second commit reverts the changes in dist in the first. Right? Do you rather want a single commit containing all changes? In that case, how do I do that? A new pull-request?

**Comment by shellscape** [comment]
I squashed them, no worries.

**Comment by shellscape** [comment]
Was gonna fix #3 and get that into the latest publish, but I got thrown a curveball for today. So I published v0.1.7 with your changes. Thanks for the PR 🎉

**Pull request merged**